src/v/model/adl_serde.cc

```diff
@@ -37,6 +37,7 @@ model::ns adl<model::ns>::from(iobuf_parser& in) {
 
 void adl<model::topic_partition>::to(iobuf& out, model::topic_partition&& t) {
     auto str = ss::sstring(t.topic);
+    // NOLINTNEXTLINE(performance-move-const-arg)
```

Comment:
What's the rationale for keeping the move?

Interestingly, serialize uses `move`, not `forward`.
```c++
template<typename... T>
void serialize(iobuf& out, T&&... args) {
    (adl<std::decay_t<T>>{}.to(out, std::move(args)), ...);
}
```
Which might be a bug.

Reply:
Probably best to move `t.topic`, but that's not part of this refactoring, I guess.

Reply:
Fixed. I was confusing this with serde, which doesn't compile without an rvalue ref.